src/pages/private/FavoriteDetails.jsx

```diff
@@ -0,0 +1,48 @@
+import { useParams, useNavigate } from 'react-router-dom';
+import { useEffect } from 'react';
+import { VideosWrapper } from '../../components/styles/VideosWrapper.styled';
+import { PageContainer } from '../../components/styles/PageContainer.styled';
+import { useContext } from 'react';
+import { GlobalContext } from '../../context/GlobalContext';
+import { lightStyles } from '../../components/styles/ThemeStyles.styled';
+import { darkStyles } from '../../components/styles/ThemeStyles.styled';
+import VideoContainer from '../../components/VideoContainer';
+import SuggestedVideosContainer from '../../components/SuggestedVideosContainer';
+
+const FavoriteDetails = () => {
+  const { darkMode, favorites, isLoggedIn } = useContext(GlobalContext);
+  const navigate = useNavigate();
+  useEffect(() => {
+    if (!isLoggedIn) {
+      navigate('/log-in');
+      return;
+    }
+  }, []);
+
+  const backgroundColor = darkMode
+    ? darkStyles.backgroundColor
+    : lightStyles.backgroundColor;
+  const { id } = useParams();
+  const [video] = favorites.filter((favorite) => id === favorite.id);
```

Comment:
It think the best to do here is once you get the video id fetch its information. By doing it that way you don't need to store all the information from a video and just its ID